my_enumerable.rb

```diff
@@ -0,0 +1,91 @@
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    for item in self do
+      new_arr.push(item) if yield(item)
+    end
+    new_arr
+  end
+
+  def my_all?
+    result = true
+    for item in self do
+      return false if yield(item) == false
+    end
+    result
+  end
+
+  def my_any?
+    for item in self do
+      return true if yield(item)
+    end
+    false
+  end
+
+  def my_none?
+    for item in self do
+      return false if yield(item)
+    end
+    true
+  end
+
+  def my_count
+    counter = 0
+    for item in self
+      counter += 1 if yield(item)
+    end
+    counter
+  end
+
+  def my_map
+    new_arr = []
+    for item in self do
+      new_arr << yield(item)
+    end
+    new_arr
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an Enumerator if no block is given
```ruby
array.my_map == Enumerator # should return true
```